subiquity/tests/api/test_api.py

```diff
@@ -1632,3 +1632,21 @@ async def test_wslsetupoptions(self):
 
             resp = await inst.get(endpoint)
             self.assertFalse(resp['install_language_support_packages'])
+
+
+class TestActiveDirectory(TestAPI):
+    async def test_ad(self):
+        async with start_server('examples/simple.json') as instance:
+            endpoint = '/active_directory'
+            ad_dict = await instance.get(endpoint)
+            # Starts empty
+            self.assertIsNone(ad_dict)
+
+            # POST succeeds
+            ad_dict = {
+                'domain_name': 'ubuntu.com',
+                'admin_name': 'u',
+                'password': 'u'
+            }
+            result = await instance.post(endpoint, ad_dict)
+            self.assertIn('OK', result)
```

Comment:
```suggestion
            self.assertEqual(['OK'], result)
```

Though this depends on the answer to the return type above